infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesClientFactory.java

```diff
@@ -72,13 +160,20 @@ public KubernetesClientFactory(
    * @throws InfrastructureException if any error occurs on client instance creation.
    */
   public KubernetesClient create() throws InfrastructureException {
```

Comment:
I think this method should be removed at all. It may lead to an error because default client is not supposed to be used for performing an operation with OS cluster, while it can be used. 
As far as I see it is used only for parsing Kubernetes list. Then it makes sense to create default kubernetes client there or use `io.fabric8.kubernetes.client.utils.Serialization#unmarshal` method directly.